src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1876,7 +1932,7 @@ private static TypeSymbolWithAnnotations InferResultNullability(Conversion conve
                     // PROTOTYPE(NullableReferenceTypes): Should an explicit cast cast away
                     // outermost nullability? For instance, is `s` a `string!` or `string?`?
                     // object? obj = ...; var s = (string)obj;
-                    isNullable = operandType?.IsNullable;
+                    isNullable = (operandType is null) ? operand.IsLiteralNull() || operand.IsLiteralDefault() : operandType.IsNullable;
```

Comment:
>operand.IsLiteralNull() [](start = 57, length = 23)

Conversion to a nullable value type doesn't yield nullable reference type

Reply:
Renamed `isNullable` to `isNullableIfReferenceType`.

---
In reply to: [169829176](https://github.com/dotnet/roslyn/pull/24935#discussion_r169829176) [](ancestors = 169829176)